code/datums/components/religious_tool.dm

```diff
@@ -0,0 +1,135 @@
+
+
+/**
+  * Since all of these involve attackby, we require mega proc. Handles Invocation, Sacrificing, And Selection of Sects.
+  */
+/datum/component/religious_tool/proc/AttemptActions(datum/source, obj/item/the_item, mob/living/user)
+	if(!istype(the_item, /obj/item/storage/book/bible)) //if it's not a bible we are going to try to sacrifice.
+		if(operation_flags & RELIGION_TOOL_SACRIFICE)
+			if(!(easy_access_sect?.can_sacrifice(the_item,user)))
+				return
+			easy_access_sect.on_sacrifice(the_item,user)
+			return COMPONENT_NO_AFTERATTACK
+	else if(easy_access_sect) //we have a bible so let's see if we have a sect!
+		if(!(operation_flags & RELIGION_TOOL_INVOKE))
+			return
+		// At this point you're intentionally trying to invoke rites, so we dont need to afterattack.
+		. = COMPONENT_NO_AFTERATTACK
+		if(!easy_access_sect.rites_list)
+			to_chat(user, "<span class='notice'>Your sect doesn't have any rites to perform!")
+			return
+		if(performing_rite)
+			to_chat(user, "<span class='notice'>There is a rite currently being performed here already!")
+			return
+		var/rite_select = input(user,"Select a rite to perform!","Select a rite",null) in easy_access_sect.rites_list
+		if(!rite_select || !user.canUseTopic(parent, BE_CLOSE, FALSE, NO_TK))
+			to_chat(user,"<span class ='warning'>You cannot perform the rite at this time.</span>")
+			return
+		var/selection2type = easy_access_sect.rites_list[rite_select]
+		performing_rite = new selection2type(parent)
+		if(!performing_rite.perform_rite(user, parent))
+			QDEL_NULL(performing_rite)
+		else
+			performing_rite.invoke_effect(user, parent)
+			easy_access_sect.adjust_favor(-performing_rite.favor_cost)
+			QDEL_NULL(performing_rite)
+		return
+	else if((operation_flags & RELIGION_TOOL_SECTSELECT)) //No bible, No sect, time to select a new one!
+		. = COMPONENT_NO_AFTERATTACK //At this point you're intentionally trying to select a sect.
+		if(user.mind.holy_role != HOLY_ROLE_HIGHPRIEST)
+			to_chat(user, "<span class='warning'>You are not the high priest, and therefore cannot select a religious sect.")
+			return
+		var/list/available_options = generate_available_sects(user)
+		if(!available_options)
+			return
+
+		var/sect_select = input(user,"Select a sect (You CANNOT revert this decision!)","Select a Sect",null) in available_options
+		if(!sect_select || !user.canUseTopic(parent, BE_CLOSE, FALSE, NO_TK))
+			to_chat(user,"<span class ='warning'>You cannot select a sect at this time.</span>")
+			return
+		var/type_selected = available_options[sect_select]
+		GLOB.religious_sect = new type_selected()
+		for(var/i in GLOB.player_list)
+			if(!isliving(i))
+				continue
+			var/mob/living/am_i_holy_living = i
+			if(!am_i_holy_living.mind?.holy_role)
+				continue
+			GLOB.religious_sect.on_conversion(am_i_holy_living)
+		easy_access_sect = GLOB.religious_sect
+		SEND_SIGNAL(parent, COMSIG_RELIGION_NEWSECT)
+		if(istype(parent,/obj/structure/altar_of_gods))
```

Comment:
make this a callback, the parent can then have custom behavior when a new sect is made.